eras/conway/impl/cddl-files/conway.cddl

```diff
@@ -430,9 +430,9 @@ drep_voting_thresholds =
   ]
 
 transaction_witness_set =
-  { ? 0: [+ vkeywitness ]
+  { ? 0: nonempty_set<vkeywitness>
   , ? 1: [+ native_script ]
-  , ? 2: [+ bootstrap_witness ]
+  , ? 2: nonempty_set<bootstrap_witness>
```

Comment:
It wasn't per the plan in https://github.com/IntersectMBO/cardano-ledger/issues/3922
to have these two as nonempty_set, like this, no? 
Rather: 
```
? 0: [+ vkeywitness ] / { + $vkey => $signature }
 ? 2: [+ bootstrap_witness ]
```

Any reason for this discrepancy? To be fair, I didn't question why the plan was like that to begin with, but curious if something caused this change.

Reply:
Very good observation. We'll change the plan to use a `nonemty_set` instead. It will be much more complicated to switch it to a Map, for very little benefit. 

The idea behind a Map, is that a unique can only produce a unique signature, so one of the `s1` and `s2` will have to be an invalid signature `[(k1, s1), (k1, s2)]`.

We check those signatures in the rules, so it maybe even better to produce a predicate failure saying that the signature is invalid, rather that a deserialization failure in this case.

Reply:
Makes sense, thanks.